rust/private/rust.bzl

```diff
@@ -623,6 +629,25 @@ _common_attrs = {
         """),
         allow_files = [".rs"],
     ),
+    "stamp": attr.int(
+        doc = dedent("""\
+            Whether to encode build information into the `Rustc` action. Possible values:
+
+            - `stamp = 1`: Always stamp the build information into the `Rustc` action, even in \
+            [--nostamp](https://docs.bazel.build/versions/main/user-manual.html#flag--stamp) builds. \
+            This setting should be avoided, since it potentially kills remote caching for the target and \
+            any downstream actions that depend on it.
+
+            - `stamp = 0`: Always replace build information by constant values. This gives good build result caching.
+
+            - `stamp = -1`: Embedding of build information is controlled by the \
+            [--[no]stamp](https://docs.bazel.build/versions/main/user-manual.html#flag--stamp) flag.
+
+            Stamped targets are not rebuilt unless their dependencies change.
```

Comment:
Could you please add an example situation to give more clarity into the impact of this behavior: For example if a `rust_library` is stamped, and a `rust_binary` depends on that library, the stamped library won't be rebuilt when we change sources of the `rust_binary`. This is different from how [`cc_library.linkstamps`](https://docs.bazel.build/versions/main/be/c-cpp.html#cc_library.linkstamp) behaves.

Reply:
Correct, each target does it's own stamping, if there are no updates to the `rust_library`, it will not be rebuilt and will maintain the stamp it was cached with. If that's not desirable, stable stamping can always be used to generate the `rustc_env_files` targets so both the library and binary are guaranteed to have the same stamp. What would you like in terms of an example? Just some more documentation?

Reply:
Oh yeah I'm sorry for not being clear in my comment. I was merely asking for more documentation (everything from the colon to the end of my comment was a suggestion for added documentation - and that was absolutely not clear from the comment :).